### DISCUSSION TITLE
stock split extraction

### DISCUSSION THREAD

**Discussion by mrx23dot** [manusimidt/py-xbrl#51] [Q&A] [answered]
Taking a simple TSLA report:

https://www.sec.gov/ix?doc=/Archives/edgar/data/0001318605/000156459020047486/tsla-10q_20200930.htm

There are 10 facts about one stock split with tag: StockholdersEquityNoteStockSplit, of course all with different timestamps.
How do I actually tell which is the correct date for the 1:5 stock split?

Correct fact
![correct](https://user-images.githubusercontent.com/4163396/127558887-55893fa7-5dfb-4c61-9ec5-ad4aa0287315.jpg)

One of incorrect one:
![incorrect](https://user-images.githubusercontent.com/4163396/127558904-4a703096-df6f-42d1-84cd-c630b8fea0d8.jpg)

Cheers

**Comment by mrx23dot** [comment]
SEC told me it's in 8-k instead.

**Comment by manusimidt** [comment, marked as answer]
I don't really know why one of the facts has another date (30.09.2019) than the other 4 Facts (30.09.2020). 
I think this is an error in the document (Although a 10-Q of such a big company like Tesla should not contain such errors...).

If you look at the place where the fact has been tagged in the instance document, you will also notice that the text written here and the XBRL tagging do not fit together. 🤔
![image](https://user-images.githubusercontent.com/29599104/127872503-03ab08d5-4138-463a-9bf7-ee2a928f363a.png)

Please mark this comment as answer if it helps :)

**Reply by mrx23dot** [reply]
Yeah, some companies don't even mention stock splits in 10K/Qs, I think I need to build up a look up table from 8Ks.
